pkg/kubelet/kubeletconfig/checkpoint/checkpoint.go

```diff
@@ -30,8 +31,8 @@ import (
 type Checkpoint interface {
 	// UID returns the UID of the config source object behind the Checkpoint
 	UID() string
-	// Parse parses the checkpoint into the internal KubeletConfiguration type
-	Parse() (*kubeletconfig.KubeletConfiguration, error)
+	// Parse extracts the KubeletConfiguration from the checkpoint, applies defaults, and converts to the internal type
+	Parse(kubeletCodecs *serializer.CodecFactory) (*kubeletconfig.KubeletConfiguration, error)
```

Comment:
Why change this signature? Can the checkpoint implementation hold a codec internally?